### PR TITLE
AssParser: Handle QNaN mesh tangents

### PR DESCRIPTION
**Pull request by badosu**

### CODE REVIEW

**Review by Beherith — APPROVED** [review]
I think the error logs very verbosely (and correctly!) convey the information that the piece has degenerate geometry. The only thing I would add, that would be more 'readable' to modelers, is printing the number of vertices for that face, as modelers do know that < 3 vertices for a face is degenerate.

**Comment by badosu** [comment]
I changed the code so it just does not assign if QNAN, similar to how already is done on aiNormal. Seems to have no negative implications on both release and debug builds.

I also made it log as INFO instead of ERROR since this is not critical information (release builds won't log it).

I also was not able to find a way to tell how many vertices the face for vertexIndex has, if someone can tell me how I can add to the message.

File: rts/Rendering/Models/AssParser.cpp [hunk 1]
```diff
@@ -510,16 +510,29 @@ void CAssParser::LoadPieceGeometry(SAssPiece* piece, const S3DModel* model, cons
 			// vertex normal
 			const aiVector3D& aiNormal = mesh->mNormals[vertexIndex];
 
-			if (!IS_QNAN(aiNormal))
+			if (IS_QNAN(aiNormal)) {
+				LOG_SL(LOG_SECTION_PIECE, L_DEBUG, "Malformed normal (model->name=\"%s\" piece->name=\"%s\" vertexIndex=%d x=%f y=%f z=%f)", model->name.c_str(), piece->name.c_str(), vertexIndex, aiNormal.x, aiNormal.y, aiNormal.z);
```
Comment (lhog) [inline rts/Rendering/Models/AssParser.cpp:514]: Let's add some sane normal here, for example `vertex.normal = float3{0.0f, 1.0f, 0.0f}`

File: rts/Rendering/Models/AssParser.cpp [hunk 2]
```diff
@@ -510,16 +510,29 @@ void CAssParser::LoadPieceGeometry(SAssPiece* piece, const S3DModel* model, cons
...
 
 			// vertex tangent, x is positive in texture axis
 			if (mesh->HasTangentsAndBitangents()) {
 				const aiVector3D& aiTangent = mesh->mTangents[vertexIndex];
 				const aiVector3D& aiBitangent = mesh->mBitangents[vertexIndex];
 
-				vertex.sTangent = (aiVectorToFloat3(aiTangent)).SafeANormalize();
-				vertex.tTangent = (aiVectorToFloat3(aiBitangent)).SafeANormalize();
+				if (IS_QNAN(aiTangent.x) || IS_QNAN(aiTangent.y) || IS_QNAN(aiTangent.z)) {
+					LOG_SL(LOG_SECTION_PIECE, L_INFO, "Malformed tangent (model->name=\"%s\" piece->name=\"%s\" vertexIndex=%d x=%f y=%f z=%f)", model->name.c_str(), piece->name.c_str(), vertexIndex, aiTangent.x, aiTangent.y, aiTangent.z);
```
Comment (lhog) [inline rts/Rendering/Models/AssParser.cpp:525]: `vertex.sTangent = float3{1.0f, 0.0f, 0.0f}`

File: rts/Rendering/Models/AssParser.cpp [hunk 3]
```diff
@@ -510,16 +510,29 @@ void CAssParser::LoadPieceGeometry(SAssPiece* piece, const S3DModel* model, cons
...
+				} else {
+					vertex.sTangent = (aiVectorToFloat3(aiTangent)).SafeANormalize();
+				}
+
+				if (IS_QNAN(aiBitangent.x) || IS_QNAN(aiBitangent.y) || IS_QNAN(aiBitangent.z)) {
+					LOG_SL(LOG_SECTION_PIECE, L_INFO, "Malformed bitangent (model->name=\"%s\" piece->name=\"%s\" vertexIndex=%d x=%f y=%f z=%f)", model->name.c_str(), piece->name.c_str(), vertexIndex, aiBitangent.x, aiBitangent.y, aiBitangent.z);
```
Comment (lhog) [inline rts/Rendering/Models/AssParser.cpp:531]: `vertex.tTangent = vertex.normal.cross(vertex.sTangent);`

**Comment by badosu** [comment]
@lhog Done

**Pull request merged**